tests/models/bert_japanese/test_tokenization_bert_japanese.py

```diff
@@ -172,6 +174,298 @@ def test_mecab_tokenizer_no_normalize(self):
             ["ｱｯﾌﾟﾙストア", "で", "iPhone", "８", "が", "発売", "さ", "れ", "た", "　", "。"],
         )
 
+    def test_pickle_sudachi_tokenizer(self):
+        tokenizer = self.tokenizer_class(self.vocab_file, word_tokenizer_type="sudachi")
+        self.assertIsNotNone(tokenizer)
+
+        text = "こんにちは、世界。\nこんばんは、世界。"
+        tokens = tokenizer.tokenize(text)
+        self.assertListEqual(tokens, ["こんにちは", "、", "世界", "。", "こん", "##ばんは", "、", "世界", "。"])
+        self.assertListEqual(tokenizer.convert_tokens_to_ids(tokens), [3, 12, 10, 14, 4, 9, 12, 10, 14])
+
+        filename = os.path.join(self.tmpdirname, "tokenizer.bin")
+        with open(filename, "wb") as handle:
+            pickle.dump(tokenizer, handle)
+
+        with open(filename, "rb") as handle:
+            tokenizer_new = pickle.load(handle)
+
+        tokens_loaded = tokenizer_new.tokenize(text)
+
+        self.assertListEqual(tokens, tokens_loaded)
+
+    def test_sudachi_tokenizer_small(self):
+        tokenizer = SudachiTokenizer(sudachi_dict_type="small")
+
+        self.assertListEqual(
+            tokenizer.tokenize(" \tｱｯﾌﾟﾙストアでiPhone８ が  \n 発売された　。  "),
+            [
+                " ",
+                "\t",
+                "アップル",
+                "ストア",
+                "で",
+                "iPhone",
+                "8",
+                " ",
+                "が",
+                " ",
+                " ",
+                "\n ",
+                "発売",
+                "さ",
+                "れ",
+                "た",
+                " ",
+                "。",
+                " ",
+                " ",
+            ],
+        )
+
+    def test_sudachi_tokenizer_core(self):
+        tokenizer = SudachiTokenizer(sudachi_dict_type="core")
+
+        self.assertListEqual(
+            tokenizer.tokenize(" \tｱｯﾌﾟﾙストアでiPhone８ が  \n 発売された　。  "),
+            [
+                " ",
+                "\t",
+                "アップル",
+                "ストア",
+                "で",
+                "iPhone",
+                "8",
+                " ",
+                "が",
+                " ",
+                " ",
+                "\n ",
+                "発売",
+                "さ",
+                "れ",
+                "た",
+                " ",
+                "。",
+                " ",
+                " ",
+            ],
```

Comment:
Put all of this in one line and deactivate the formatter (with a line `# fmt: off` before and a line `# fmt: on` after.